### ISSUE TITLE
Qoe Metrics Reporting - Combination of M1 and M5 calls leads to segmentation fault

### ISSUE THREAD

**Issue by dsilhavy** [5G-MAG/rt-5gms-application-function#149]
## Description
When testing the "end to end" setup for QoE Metrics Reporting as outlined below, a segmentation fault occurs.
### 1. Create a provisioning session (M1).
Example response after creation:

````json
{
    "provisioningSessionId": "c935fe48-e755-41ee-ad3f-5937a1783bfe",
    "provisioningSessionType": "DOWNLINK",
    "aspId": "aspId",
    "appId": "appId",
    "serverCertificateIds": [],
    "metricsReportingConfigurationIds": []
}
````

### 2. Create a Metrics Reporting Configuration (M1)

Example payload:
````json
{
    "scheme": "scheme2",
    "dataNetworkName": "dataNetworkName2",
    "reportingInterval": 10,
    "samplePercentage": 5,
    "urlFilters": ["a","b"],
    "samplingPeriod": 5,
    "metrics": ["metrica", "metricb"]
}
````

### 3. Query the Service Access Information (M5)
Example response:

````json
{
    "provisioningSessionId": "c935fe48-e755-41ee-ad3f-5937a1783bfe",
    "provisioningSessionType": "DOWNLINK",
    "streamingAccess": {},
    "clientMetricsReportingConfigurations": [
        {
            "metricsReportingConfigurationId": "f2684708-e755-41ee-ad3f-5937a1783bfe",
            "serverAddresses": [
                "http://192.168.2.7:7778/3gpp-m5/v2/"
            ],
            "scheme": "scheme2",
            "dataNetworkName": "dataNetworkName2",
            "reportingInterval": 10,
            "samplePercentage": 5,
            "urlFilters": [
                "a",
                "b"
            ],
            "samplingPeriod": 5,
            "metrics": [
                "metrica",
                "metricb"
            ]
        }
    ],
    "networkAssistanceConfiguration": {
        "serverAddresses": [
            "http://192.168.2.7:7778/3gpp-m5/v2/"
        ]
    }
}
````

### 4. Query the QoE Metrics Reporting Configuration (M1)
The AF crashes with a segmentation fault

**Comment by stojkovicv** [comment]
Which HTTP methods are failing at the step 4?

**Comment by dsilhavy** [comment]
> Which HTTP methods are failing at the step 4?

"Query" means doing a `GET` request. Probably the other ones fail as well, this needs to be tested.

**Comment by stojkovicv** [comment]
Few additional notes:

Quering M1 Metrics will fail **only** if SAI GET is conducted before. Sometimes it comes with `200 OK` but returns invalid data:

```
{
    "metricsReportingConfigurationId":	"�",
    "scheme":	"p",
    "dataNetworkName":	"�",
    "reportingInterval": 10,
    "samplePercentage": 25.5,
    "urlFilters": [],
    "samplingPeriod": 0,
    "metrics": []
}
```

Backtrace states that a fault point for M1 Metrics GET is invokation of `msaf_api_metrics_reporting_configuration_convertToJSON()`.

SAI only reads the values from the existing Metrics Objects (more precisely from 
`provisioning_session->metrics_reporting_map` ), and it shouldn't affect existing internal data.

For, still unknown reason, internal metrics data seems damaged after SAI related HTTP method.

**Comment by dsilhavy** [comment]
Potentially addressed by #150

**Comment by davidjwbbc** [comment]
@dsilhavy I'll leave this open until you've checked that my double-free fix corrects this issue for you.

**Comment by dsilhavy** [comment]
Confirmed this is fixed with #150
